File: reference/src/com/db4odoc/diagnostics/IndexDiagListener.java
/* Copyright (C) 2004 - 2007 db4objects Inc. http://www.db4o.com */
package com.db4odoc.diagnostics;

import com.db4o.diagnostic.*;

public class IndexDiagListener  implements DiagnosticListener
{
	   public void onDiagnostic(Diagnostic d) {
		   if (d.getClass().equals(LoadedFromClassIndex.class)){
	        System.out.println(d.toString());
		   }
	    }
}

File: reference.net/db4o-reference-chapters/src/Db4objects.Db4odoc/Diagnostics/IndexDiagListener.cs
/* Copyright (C) 2004 - 2007 db4objects Inc. http://www.db4o.com */
using Db4objects.Db4o.Diagnostic;

namespace Db4objects.Db4odoc.Diagnostics
{
	public class IndexDiagListener: DiagnosticToConsole
	{
		override public void OnDiagnostic(Db4objects.Db4o.Diagnostic.IDiagnostic d) 
		{
            if (d.GetType().Equals(typeof(Db4objects.Db4o.Diagnostic.LoadedFromClassIndex)))
            {
				System.Console.WriteLine(d.ToString());
			}
		}
	}
}